packages/doenetml-worker-rust/lib-doenetml-core/src/new_core/graph_based_core.rs

```diff
@@ -208,4 +526,191 @@ impl Core {
             }
         }
     }
+
+    /// Add component `child_idx` as the first child of `parent_idx`.
+    fn prepend_child_to_structure_graph(
```

Comment:
This is adding a child to the component, not a "graph child". The name should emphasize this. Maybe `prepend_component_child`. I don't know if `structure_graph` needs to be mentioned, since there is no other place to do stuff with children.